rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2252,50 +2391,109 @@ extern "C"
       const char * topic_name,
       size_t * count)
     {
-        (void)node;
-        (void)topic_name;
-        (void)count;
-        RMW_SET_ERROR_MSG("not implemented");
-        return RMW_RET_ERROR;
+	//safechecks
+	
+	if(!node){
+		RMW_SET_ERROR_MSG("null node handle");
+		return RMW_RET_ERROR;
+	}	
+	//Get participant pointer from node
+        if(node->implementation_identifier != eprosima_fastrtps_identifier)
+        {
+            RMW_SET_ERROR_MSG("node handle not from this implementation");
+            return RMW_RET_ERROR;
+        }
+
+        CustomParticipantInfo* impl = static_cast<CustomParticipantInfo*>(node->data);
+
+	std::map<std::string,std::set<std::string>>unfiltered_topics;
```

Comment:
Does the `set` here imply that topics with the same type are only counted once here? This could be the reason for #127.